src/screens/CreateCommunity/index.js

```diff
@@ -0,0 +1,327 @@
+  const navigation = useNavigation();
+  const [users, dispatch] = React.useContext(Context).users;
+  const [name, setName] = React.useState('');
+  const [typeFetch, setTypeFetch] = React.useState('');
+
+  const verifyNameDebounce = React.useCallback(
+    _.debounce(async (text) => {
+      if (text?.length < MINIMUM_NAME_LENGTH) return setTypeFetch('min');
+      if (text?.length > MAXIMUM_NAME_LENGTH) return setTypeFetch('max');
+      const result = await verifyCommunityName(text);
+      if (!result.available) {
+        return setTypeFetch('notavailable');
+      }
+
+      return setTypeFetch('available');
+    }, 500),
+    []
+  );
+
+  const checkName = async (v) => {
+    verifyNameDebounce.cancel();
+    const value = v.replace(/[^a-zA-Z0-9-_]/g, '');
+    setTypeFetch('typing');
+    setName(value);
+    if (value.length <= MAXIMUM_NAME_LENGTH) {
+      if (value.length >= MINIMUM_NAME_LENGTH) {
+        if (!Number.isNaN(v)) {
+          setTypeFetch('fetch');
+          verifyNameDebounce(value);
+        } else {
+          setTypeFetch('nan');
+        }
+      } else {
+        setTypeFetch('typing');
+      }
+    } else {
+      setTypeFetch('max');
+    }
+  };
+
+  const formatNameString = () => {
+    if (name && typeof name === 'string') {
+      let value = name.toLowerCase().replace(/[^a-z0-9-_]/g, '');
+      value = setCapitalFirstLetter(value);
+      return value;
+    }
+    return '';
+  };
+
+  const onTextBlur = () => {
+    const value = formatNameString();
+    setName(value);
+  };
+
+  // eslint-disable-next-line consistent-return
+  const next = () => {
+    if (name && name.length >= MINIMUM_NAME_LENGTH && typeFetch === 'available') {
+      navigation.replace(NavigationConstants.CREATE_POST_SCREEN, {
+        isCreateCommunity: true,
+        topic: name
+      });
+    } else {
+      if (!name) {
+        return showMessage({
+          message: 'Community name cannot be empty',
+          type: 'danger',
+          backgroundColor: COLORS.redalert
+        });
+      }
+
+      if (name.length < MINIMUM_NAME_LENGTH) {
+        return showMessage({
+          message: 'Community name min. 3 characters',
+          type: 'danger',
+          backgroundColor: COLORS.redalert
+        });
+      }
+
+      if (name.length > MAXIMUM_NAME_LENGTH) {
+        return showMessage({
+          message: 'Community name maximum 64 characters',
+          type: 'danger',
+          backgroundColor: COLORS.redalert
+        });
+      }
+
+      if (typeFetch === 'notavailable') {
+        return showMessage({
+          message: 'A community with this name already exists',
+          type: 'danger',
+          backgroundColor: COLORS.redalert
+        });
+      }
+
+      if (typeFetch === 'nan') {
+        return showMessage({
+          message: 'Community name cannot be just a number',
+          type: 'danger',
+          backgroundColor: COLORS.redalert
+        });
+      }
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `next` function is also doing a lot of things. It's checking the name, navigating to another screen, and showing error messages. This could be broken down into smaller functions for better readability and maintainability.

```diff
-  const next = () => {
-    if (name && name.length >= MINIMUM_NAME_LENGTH && typeFetch === 'available') {
-      navigation.replace(NavigationConstants.CREATE_POST_SCREEN, {
-        isCreateCommunity: true,
-        topic: name
-      });
-    } else {
-      if (!name) {
-        return showMessage({
-          message: 'Community name cannot be empty',
-          type: 'danger',
-          backgroundColor: COLORS.redalert
-        });
-      }
-
-      if (name.length < MINIMUM_NAME_LENGTH) {
-        return showMessage({
-          message: 'Community name min. 3 characters',
-          type: 'danger',
-          backgroundColor: COLORS.redalert
-        });
-      }
-
-      if (name.length > MAXIMUM_NAME_LENGTH) {
-        return showMessage({
-          message: 'Community name maximum 64 characters',
-          type: 'danger',
-          backgroundColor: COLORS.redalert
-        });
-      }
-
-      if (typeFetch === 'notavailable') {
-        return showMessage({
-          message: 'A community with this name already exists',
-          type: 'danger',
-          backgroundColor: COLORS.redalert
-        });
-      }
-
-      if (typeFetch === 'nan') {
-        return showMessage({
-          message: 'Community name cannot be just a number',
-          type: 'danger',
-          backgroundColor: COLORS.redalert
-        });
-      }
-    }
-  };
+  const navigateToCreatePost = () => {
+    navigation.replace(NavigationConstants.CREATE_POST_SCREEN, {
+      isCreateCommunity: true,
+      topic: name
+    });
+  };
+
+  const showError = (message) => {
+    showMessage({
+      message,
+      type: 'danger',
+      backgroundColor: COLORS.redalert
+    });
+  };
+
+  const next = () => {
+    if (!name) {
+      return showError('Community name cannot be empty');
+    }
+
+    const lengthCheck = checkLength(name);
+    if (lengthCheck === 'min') {
+      return showError('Community name min. 3 characters');
+    }
+    if (lengthCheck === 'max') {
+      return showError('Community name maximum 64 characters');
+    }
+
+    if (typeFetch === 'notavailable') {
+      return showError('A community with this name already exists');
+    }
+
+    if (checkNumeric(name)) {
+      return showError('Community name cannot be just a number');
+    }
+
+    if (typeFetch === 'available') {
+      navigateToCreatePost();
+    }
+  };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->